src/dataframe/dataframe.jl

```diff
@@ -499,6 +499,10 @@ Base.setproperty!(df::DataFrame, col_ind::Symbol, v::AbstractVector) =
     (df[!, col_ind] = v)
 Base.setproperty!(df::DataFrame, col_ind::AbstractString, v::AbstractVector) =
     (df[!, col_ind] = v)
+Base.setproperty!(::AbstractDataFrame, ::Symbol, ::Any) =
+    throw(ArgumentError("it is only allowed to pass a vector as a column of a DataFrame"))
+Base.setproperty!(::AbstractDataFrame, ::AbstractString, ::Any) =
+    throw(ArgumentError("it is only allowed to pass a vector as a column of a DataFrame"))
```

Comment:
Would it make sense to point people to `df[:, col] .= v`? This is a very common mistake.

Reply:
Right. I will have to add more methods then. For `DataFrame` it should be `df!, col] .= v` (replacing a column), and for `SubDataFrame` either it should keep erroring or use `df:, col] .= v` (updating in place).

What do you think would be a better hint for `SubDataFrame`?

Reply:
`SubDataFrame` doesn't support `setproperty!` at all, right? So we can point to both `sdf[:, col] = v` and  `sdf[!, col] .= v`, noting that these are in-place.

BTW, methods for `AbstractDataFrame` should probably got to abstractdataframe.jl (or only define them for `DataFrame` if we don't define working methods for `AbstractDataFrame`).